src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -287,26 +356,28 @@ function toTree(node: ?Fiber) {
     case ClassComponent:
       return {
         nodeType: 'component',
-        type: node.type,
+        type: publicType(node),
         props: {...node.memoizedProps},
         instance: node.stateNode,
-        rendered: toTree(node.child),
+        rendered: childrenToTree(node.child),
       };
+    case Fragment: // 10
+      return childrenToTree(node.child);
     case FunctionalComponent: // 1
       return {
         nodeType: 'component',
-        type: node.type,
+        type: publicType(node),
         props: {...node.memoizedProps},
         instance: null,
-        rendered: toTree(node.child),
+        rendered: childrenToTree(node.child),
       };
     case HostComponent: // 5
       return {
         nodeType: 'host',
         type: node.type,
         props: {...node.memoizedProps},
         instance: null, // TODO: use createNodeMock here somehow?
-        rendered: nodeAndSiblingsArray(node.child).map(toTree),
+        rendered: flatten(nodeAndSiblingsArray(node.child).map(toTree)),
```

Comment:
Can we make this faster? This is super inefficient atm. We have plenty of examples in the codebase of linked list tree traversal without the intermediate arrays.

But why are only host components flattened and not other types?